dexbot/strategies/king_of_the_hill.py

```diff
@@ -0,0 +1,376 @@
+
+        self.log.info("{} initialized.".format(STRATEGY_NAME))
+
+    def maintain_strategy(self, *args):
+        """ Strategy main logic
+        """
+        delta = datetime.now() - self.last_check
+        # Only allow to check orders whether minimal time passed
+        if delta < timedelta(seconds=self.min_check_interval):
+            return
+
+        self.calc_order_prices()
+
+        if self.orders:
+            self.check_orders()
+        else:
+            self.place_orders()
+
+        self.last_check = datetime.now()
+
+    def check_orders(self):
+        """ Check whether own orders needs intervention
+        """
+        orders_to_delete = []
+        for stored_order in self.orders:
+            order = self.get_order(stored_order['id'])
+            order_type = self.get_order_type(stored_order)
+
+            if order:
+                is_partially_filled = self.is_partially_filled(order, threshold=0.8)
+                if is_partially_filled:
+                    # If own order filled too much, replace it with new order
+                    # TODO: check cancel_orders() return value after #467 fix
+                    self.log.info('Own {} order filled too much, resetting'.format(order_type))
+                    self.cancel_orders(order)
+                    orders_to_delete.append(stored_order['id'])
+                    self.place_order(order_type)
+                # Check if someone put order above ours or beaten order was canceled
+                elif ((order_type == 'buy' and (not self.get_order(self.beaten_buy_order) or
+                        stored_order['price'] < self.buy_price)) or
+                        (order_type == 'sell' and (not self.get_order(self.beaten_sell_order) or
+                            # TODO: remove price invert after #475 fix
+                            stored_order['price'] ** -1 > self.sell_price))):
+                    self.log.debug('Moving {} order'.format(order_type))
+                    self.cancel_orders(order)
+                    orders_to_delete.append(stored_order['id'])
+                    self.place_order(order_type)
+            # Own order is not there
+            else:
+                self.log.info('Own {} order filled, placing a new one'.format(order_type))
+                orders_to_delete.append(stored_order['id'])
+                self.place_order(order_type)
+
+        self.orders = [o for o in self.orders if o['id'] not in orders_to_delete]
+
+    def get_order_type(self, order):
+        """ Determine order type and return it as a string
+        """
+        if self.is_buy_order(order):
+            return 'buy'
+        else:
+            return 'sell'
+
+    def calc_order_prices(self):
+        """ Calculate prices of our orders
+        """
+        # Obtain orderbook orders excluding our orders
+        market_orders = self.get_market_orders(depth=100)
+        own_orders_ids = [o['id'] for o in self.get_own_orders]
```

Comment:
Same goes here for the variable name as well as other places. I won't comment each.